src/lua_console.h

```diff
@@ -0,0 +1,44 @@
+#ifndef LUA_CONSOLE_H
+#define LUA_CONSOLE_H
+
+#include "output.h"
+#include "input.h"
+
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+
+struct WINDOW;
+
+class lua_console {
+    public:
+        lua_console();
+        void dispose();
+        void run();
+    private:
+        const int width = TERMX;
+        const int lines = 10;
+
+        WINDOW *cWin;
+        WINDOW *iWin;
+
+        std::vector<std::pair<std::string, nc_color>> text_stack;
+        std::string get_input();
+        void print( std::string text );
+        void draw();
+
+        bool done = false;
+
+        int scroll = 0;
+
+        void quit();
+        void scroll_up();
+        void scroll_down();
+
+        instance_invokable<lua_console> quit_callback = instance_invokable<lua_console>(this, &quit);
```

Comment:
although, I'm some trouble defining my map parameter: ```std::map<long, std::function *>``` doesn't work because we don't have a type argument. Does ```std::function``` extend from something like my invokable interface too?

Otherwise the the parameter would be ```std::map<long, std::function<void(const lua_console&)>>``` meaning we could only use it for ```lua_console``` members.